.github/workflows/publish.yml

```diff
@@ -58,6 +58,12 @@ jobs:
       - name: Build
         run: ./gradlew build
 
+      - name: Upload Arrow version file
+        uses: actions/upload-artifact@v2
+        with:
+          name: arrow-version
+          path: build/semver/version.txt
```

Comment:
This step can read the file and add the version to the next job:

https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idoutputs

This suggestion is based on the premise that uploading/downloading the file should be slower than just reading the file and setting the version to the output. If the time is the same then it doesn't matter.

Reply:
I think it's a good suggestion. I'll fix it